### PR TITLE
refactor(vector): remove layer dependency for vector API

### PR DESCRIPTION
**Pull request by XuNeo**
### Description of the feature or fix

Align vector draw API with other draw method by adding `layer` and `coords` parameter. `coords` is not used for now.
```c
void lv_draw_vector(lv_layer_t * layer, lv_vector_dsc_t * dsc, const lv_area_t *coords);

/*Similar to */
void lv_draw_image(lv_layer_t * layer, const lv_draw_image_dsc_t * dsc, const lv_area_t * coords);
void lv_draw_label(lv_layer_t * layer, const lv_draw_label_dsc_t * dsc, const lv_area_t * coords);


```

Normally we prepare the vector path firstly and draw later when needed. Layer is only available the moment to draw it.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/draw/lv_draw_vector.c [hunk 1]
```diff
@@ -825,15 +812,29 @@ void lv_vector_clear_area(lv_vector_dsc_t * dsc, const lv_area_t * rect)
...
-    lv_layer_t * layer = dsc->layer;
+    lv_ll_t * task_list = dsc->tasks.task_list;
+    _lv_vector_draw_task * task = _lv_ll_get_head(task_list);
+    _lv_vector_draw_task * next_task = NULL;
+
+    while(task != NULL) {
+        next_task = _lv_ll_get_next(task_list, task);
+
+        if(coords) {
+            lv_matrix_translate(&task->dsc.matrix, coords->x1, coords->y1);
```
Comment (onecoolx) [inline src/draw/lv_draw_vector.c:829]: coords  is clip area , not need transform by global matrix

File: src/draw/lv_draw_vector.c [hunk 2]
```diff
@@ -825,15 +812,29 @@ void lv_vector_clear_area(lv_vector_dsc_t * dsc, const lv_area_t * rect)
...
+        }
+
+        _lv_area_intersect(&task->dsc.scissor_area, &task->dsc.scissor_area, &layer->_clip_area);
```
Comment (onecoolx) [inline src/draw/lv_draw_vector.c:832]: it is need intersect with coords ?

**Review by onecoolx — APPROVED** [review]

File: src/draw/lv_draw_vector.c [hunk 3]
```diff
@@ -825,15 +812,29 @@ void lv_vector_clear_area(lv_vector_dsc_t * dsc, const lv_area_t * rect)
...
+
+        task = next_task;
+    }
 
-    lv_draw_task_t * t = lv_draw_add_task(layer, &(layer->_clip_area));
+    lv_draw_task_t * t = lv_draw_add_task(layer, &layer->_clip_area);
```
Comment (kisvegabor) [inline src/draw/lv_draw_vector.c:837]: Here `coords` (the total area of the drawing) shall be used similarly to e.g. [`lv_draw_label()`](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_draw_label.c#L85) 
```suggestion
    lv_draw_task_t * t = lv_draw_add_task(layer, coords);
```

**Review by kisvegabor — COMMENTED** [review]
I've found an optimization opportunity: In [`lv_draw_add_task`](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_draw.c#L89-L92) we could have:
```c
    new_task->_real_area = *coords;
    new_task->clip_area = layer->_clip_area;
   _lv_area_intersect(new_task->area, new_task->clip_area, &new_task->real_area);
```

Use `real_area` during drawing and `area` in [`is_independent`](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_draw.c#L405C13-L405C27). This way when we look for independent areas, the clipped areas will be considered which can be smaller,  so it's more likely that we find independent areas.

**Pull request closed without merging**